torch/csrc/jit/frontend/ir_emitter.cpp

```diff
@@ -3008,6 +3009,100 @@ struct to_ir {
         return std::make_shared<SimpleValue>(
             emitListComprehension(lc, type_hint));
       }
+      case prim::dict: {
+        // Set the type
+        TypePtr type = type_hint;
+        if (type_hint) {
+          if (!type_hint->cast<DictType>()) {
+            throw ErrorReport(apply.range())
+                << "Expected dict type annotation for dict(), found "
+                << type_hint->repr_str();
+          }
+        } else {
+          // Otherwise, use the default of Dict[str, Tensor]. If we have an
+          // iterable in apply.inputs(), then this default type will be
+          // overridden when the iterable is emitted
+          type = DictType::create(StringType::get(), TensorType::get());
+        }
+
+        // If we have an empty call to dict()
+        if (apply.inputs().empty() && apply.attributes().empty()) {
+          return std::make_shared<SimpleValue>(
+              graph
+                  ->insertNode(graph->createDict(
+                      type->expect<DictType>()->getKeyType(),
+                      type->expect<DictType>()->getValueType(),
+                      {},
+                      {}))
+                  ->output());
+        }
+
+        // We can't feasibly register all possible key x value
+        // combinations of new prim ops for the case that we use the
+        // constructor with a dict literal. It makes much more sense
+        // to transform the dict literal into a list of tuples so that
+        // we can use the existing constructors
```

Comment:
This comment I think could be rephrased. `We can't feasibly register all possible key x value combinations`, while  this is true, it is no different than the existing constructors, which use typevars so that the ops can be generic over input types. 

Maybe -> `transform into list of tuples to reuse existing  constructors` ? or something along those lines